cmd/tink-worker/main.go

```diff
@@ -2,18 +2,56 @@ package main
 
 import (
 	"fmt"
+	"math/rand"
 	"os"
+	"time"
 
+	"github.com/tinkerbell/tink/cmd/tink-worker/client"
 	"github.com/tinkerbell/tink/cmd/tink-worker/cmd"
+	"github.com/tinkerbell/tink/protos/workflow"
+	"google.golang.org/grpc"
 )
 
 func main() {
-	cmdlineFlags, err := cmd.CollectCmdlineFlags(os.Args[1:])
+	// parse and validate command-line flags and required env vars
+	flagEnvSettings, err := cmd.CollectFlagEnvSettings(os.Args[1:])
 	if err != nil {
 		fmt.Println("Error:", err)
 		os.Exit(1)
 	}
 
-	// Remove me: this is just here to prevent linter issues
-	fmt.Printf("Version flag is %v\n", cmdlineFlags.Version)
+	// establish connection to tink server
+	creds, err := client.ObtainTinkServerCreds(flagEnvSettings.TinkServerURL)
```

Comment:
Hmm....might consider moving this into the for-loop. Right now, this gives us one shot at getting the creds. An intermittent or slow network issue might be resolved on its own and it would be nice to try again.

Reply:
I'm not sure if there is an elegant way you had in mind to be able to do this in one loop? Unless it would be two nested loops? Or I could just add a separate loop for the `ObtainTinkServerCreds()` step.